### PR TITLE
Obliviate ALL_TENSORTYPES and ALL_TENSORTYPES2.

### LINKED ISSUE #71096: Obliviate ALL_TENSORTYPES and ALL_TENSORTYPES2
### 🚀 The feature, motivation and pitch

Usage of these two variables in tests is outdated. One should just write in the dtypes in question or use one of the ATen macro dispatch list functions (like `get_all_fp_dtypes()`). 

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by khushi-411**
Hi,
The PR fixes #71096.  It aims to scan all the test files and replace ` ALL_TENSORTYPES` and `ALL_TENSORTYPES2` with `get_all_fp_dtypes`.

I'm looking forward to your viewpoints!

Thanks!

cc: @janeyx99 @kshitij12345

### CODE REVIEW

**Review by anjali411 — APPROVED** [review]
LGTM! cc. @jbschlosser to take a look!

File: test/test_nn.py [hunk 1]
```diff
@@ -13051,7 +13051,7 @@ def test_ConvTranspose2d_large_output_padding(self, device, dtype):
 
     @onlyCUDA
     @tf32_on_and_off(0.01)
-    @dtypes(*ALL_TENSORTYPES)
+    @dtypes(*get_all_fp_dtypes(include_bfloat16=False))
```
Comment (mruberry) [inline test/test_nn.py:13054]: This is close but not exactly what we want. A few things:

- don't use the `get_all` macros, either
- preserve the bfloat16 dtype when testing on ROCm

The dtypes should either be listed explicitly or you can use one of the functions styled after the ATen dispatch macros which are imported from here: https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_dtype.py and defined like this: 

https://github.com/pytorch/pytorch/blob/4f35b9144cd3b84779139f8946d579b52bf4c1e0/torch/testing/_legacy.py#L64

Comment (khushi-411) [reply test/test_nn.py]: > A few things:
> 
> - don't use the get_all... macros, either
> - preserve the bfloat16 dtype when testing on ROCm

Sure, I will keep it in mind. Thanks :)

> The dtypes should either be listed explicitly or you can use one of the functions styled after the ATen dispatch macros

Thanks for the hint. I've updated the files. Will you please take a look?

File: test/test_nn.py [hunk 2]
```diff
@@ -15587,7 +15587,7 @@ def test_embedding_backward(self, device, dtype):
         self.assertEqual(embedding.weight.grad._indices(), tensorTwice)
         self.assertEqual(embedding.weight.grad._values(), onesTwice)
 
-    @dtypesIfCUDA(*ALL_TENSORTYPES2)
+    @dtypesIfCUDA(*get_all_fp_dtypes(include_bfloat16=AMPERE_OR_ROCM))
```
Comment (mruberry) [inline test/test_nn.py:15590]: This would just be `include_bfloat16=TEST_WITH_ROCM`, right? The `TEST_WITH_ROCM` variable may need to be imported.

**Review by mruberry — APPROVED** [review]
Thanks @khushi-411!

File: test/test_nn.py [hunk 3]
```diff
@@ -15587,7 +15587,7 @@ def test_embedding_backward(self, device, dtype):
         self.assertEqual(embedding.weight.grad._indices(), tensorTwice)
         self.assertEqual(embedding.weight.grad._values(), onesTwice)
 
-    @dtypesIfCUDA(*ALL_TENSORTYPES2)
+    @dtypesIfCUDA(*get_all_fp_dtypes(include_bfloat16=TEST_WITH_ROCM))
```
Comment (kshitij12345) [inline test/test_nn.py:15590]: Do we want to keep using these `get_all_*` as mentioned in the previous comment.

cc: @mruberry @khushi-411

Comment (khushi-411) [reply test/test_nn.py:15590]: Hi, @kshitij12345!
I stumbled upon the link provided by @mruberry. The macros it consists were:
```
- floating_types()
- floating_types_and_half()
- floating_types_and()
- floating_and_complex_types()
- floating_and_complex_types_and()
- integral_types()
- integral_types_and()
- all_types()
- all_types_and()
- complex_types()
- all_types_and_complex()
- all_types_and_complex_and()
- all_types_and_half()
```
None of these macros includes the `torch.bfloat16` type, that we require for testing with the `TEST_WITH_ROCM` environment.

Then I searched a couple of examples that use the `bfloat16` type with the `TEST_WITH_ROCM` environment and worked accordingly.

I think the other way without using `get_all_*` can be:
```
@dtypesIfCUDA(*(floating_types_and(dtypes=*(torch.half, if TEST_WITH_ROCM: torch.bfloat16)))
```
What do you think? Shall I replace it?

Thanks :)

Comment (janeyx99) [reply test/test_nn.py:15590]: I think the `floating_types_and` would be preferable here

Comment (khushi-411) [reply test/test_nn.py]: Yes, I changed. Thanks :)

Comment (mruberry) [reply test/test_nn.py]: Ideally we'd be rid of them but it's OK for now

**Review by jbschlosser — APPROVED** [review]
Beautiful!

**Review by mruberry — APPROVED** [review]
Thanks @khushi-411!

**Comment by mruberry** [comment]
@anjali411 will merge this for you

**Comment by khushi-411** [comment]
Sure, Thanks @mruberry! :)

**Pull request closed without merging**